src/main/scala/scala/tools/refactoring/implementations/OrganizeImports.scala

```diff
@@ -559,3 +565,79 @@ abstract class OrganizeImports extends MultiStageRefactoring with TreeFactory wi
     }.transform(tree)
   }
 }
+
+class OrganizeImportsSpacer(val oiInstance: OrganizeImports) {
+  import oiInstance.global._
+  import scala.tools.nsc.interactive.Global
+
+  private def topMostClassOrModule(stats: List[Tree]): Option[ImplDef] = stats.find {
+    case classDefOrModuleDef: ImplDef => true
+    case _ => false
+  }.asInstanceOf[Option[ImplDef]]
+
+  private def linesDiffInOrigFile(classOrModule: Option[ImplDef], importt: Option[Import]): Option[Int] =
+    for {
+      cm <- classOrModule
+      imp <- importt
+      sourceFile = cm.pos.source
+      cmLine = sourceFile.offsetToLine(cm.pos.start)
+      impLine = sourceFile.offsetToLine(imp.pos.end)
+    } yield cmLine - impLine
+
+  private def lastImportInPackageScope(stats: List[Tree]): Option[Import] = topMostClassOrModule(stats).flatMap { topMost =>
+    val index = stats.indexOf(topMost)
+    stats.take(index).reverse.dropWhile { !_.isInstanceOf[Import] }.headOption.asInstanceOf[Option[Import]]
```

Comment:
thanks, `collectFirst` is much better to avoid casting


2016-03-04 10:44 GMT+01:00 Simon Schäfer <notifications@github.com>:

> In
> src/main/scala/scala/tools/refactoring/implementations/OrganizeImports.scala
> <https://github.com/scala-ide/scala-refactoring/pull/143#discussion_r55009375>
> :
>
> > +    case classDefOrModuleDef: ImplDef => true
> > +    case _ => false
> > +  }.asInstanceOf[Option[ImplDef]]
> > +
> > +  private def linesDiffInOrigFile(classOrModule: Option[ImplDef], importt: Option[Import]): Option[Int] =
> > +    for {
> > +      cm <- classOrModule
> > +      imp <- importt
> > +      sourceFile = cm.pos.source
> > +      cmLine = sourceFile.offsetToLine(cm.pos.start)
> > +      impLine = sourceFile.offsetToLine(imp.pos.end)
> > +    } yield cmLine - impLine
> > +
> > +  private def lastImportInPackageScope(stats: List[Tree]): Option[Import] = topMostClassOrModule(stats).flatMap { topMost =>
> > +    val index = stats.indexOf(topMost)
> > +    stats.take(index).reverse.dropWhile { !_.isInstanceOf[Import] }.headOption.asInstanceOf[Option[Import]]
>
> The part after reverse can be replaced by collectFirst { case imp: Import
> => imp }, couldn't it?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/scala-ide/scala-refactoring/pull/143/files#r55009375>.
>